tests/IntegrationTest.php

```diff
@@ -110,11 +113,18 @@ public function testConnectingFailsIfDnsUsesInvalidResolver()
             'dns' => $dns
         ));
 
-        $this->setExpectedException('RuntimeException');
-        Block\await($connector->connect('google.com:80'), $loop, self::TIMEOUT);
+        $result = Block\await($connector->connect('google.com:80'), $loop, self::TIMEOUT);
+
+        $this->expectExceptionMessage(
+          'DNS query for google.com failed: Unable to connect to DNS server: php_network_getaddresses: getaddrinfo failed: ' . $result
+        );
```

Comment:
This will not be executed as the above method already throws. This is also the reason why it *may* make sense to keep the original test structure (see above discussion), but I'll leave this up to you :+1: 

(see also below occurrences)

Reply:
Ping @gabriel-caruso, what's the status here?

Reply:
@clue Sorry, I didn't see that. I gonna move `expect` methods to the beginning of the methods.